compliance_checker/ioos.py

```diff
@@ -815,62 +806,114 @@ def check_single_platform(self, ds):
+
+        Examine each variable with a cf_role. For each (featureType, cf_role):
+            - (timeSeries, timeseries_id)
+            - (timeSeriesProfile, timeseries_id)
+            - (trajectory, trajectory_id)
+            - (trajectoryProfile, trajectory_id)
+            - (profile, profile_id)
+        the dimension of the variable should be 1. For datasets with
+        the "point" featureType, do nothing.
+
+
+        https://github.com/ioos/compliance-checker/issues/748#issuecomment-606659685
+
+        Parameters
+        ----------
+        ds: netCDF4.Dataset (open)
+
+        Returns
+        -------
+        list of Result objects
+        """
+
+        results = []
 
+        feature_type = getattr(ds, "featureType", "").lower()
+        if not feature_type:
+            return results
+ 
+        # loop through all variables with cf_role
+        cf_role_vars = ds.get_variables_by_attributes(cf_role=lambda x: x is not None)
+        num_cf_role_vars = len(cf_role_vars)
+
+        for var in cf_role_vars:
+            cf_role = getattr(var, "cf_role")
+            shp = var.shape[0] if len(var.shape) > 0 else 1
+ 
+            if feature_type=="timeseries" and cf_role=="timeseries_id":
+                msg = (
+                    "Dimension length of the variable with "
+                    "cf_role='timeseries_id (the 'station' dimension) "
+                    "is {dim}. Note that the IOOS profile restricts "
+                    "timeSeries datasets with multiple features to share "
+                    "the same lat/lon position (ie. to exist on the same "
+                    "platform). Datasets that include multiple platforms "
+                    "are not valid and will cause harvesting errors."
+                ).format(dim=shp)
+ 
+                _val = shp<=1 # fails if > 1
+ 
+            elif ( # even though tested condition is the same, different msg
+                     feature_type=="profile" and cf_role=="profile_id" or
+                     feature_type=="trajectory" and cf_role=="trajectory_id" or
+                     feature_type=="timeseriesprofile" and cf_role=="timeseries_id" or
+                     feature_type=="trajectoryprofile" and cf_role=="trajectory_id"
+                 ):
+
+                msg = (
+                    "Dimension length of the variable `{cf_role_var}` with "
+                    "cf_role=`{cf_role}` (the 'station/trajectory/profile' "
+                    "dimension) must be equal to 1 (it is {dim}). The IOOS "
+                    "profile restricts {feature_type} datasets to a "
+                    "single platform (ie. station/trajectory/profile) per dataset."
+                ).format(
+                    cf_role_var=var.name, cf_role=cf_role,
+                    dim=shp, feature_type=feature_type
+                )
+ 
+                _val = shp==1
+ 
+            elif feature_type=="point": # do nothing
```

Comment:
Please leave spaces around binary operators and non-keyword assignments per suggestions in PEP8.

Reply:
if you're going to enforce pep8, might be worth letting travisCI catch those violations. Likewise, @daltonkell there are tools you can use that provide a pre-commit hook to check your code for pep8 compliance.

Reply:
@Bobfrat, The thing is with CI is that a lot of the existent codebase is not necessarily PEP8 compliant already.  We have a "coding-standards" test in Travis already which at a cursory glance appears to check PEP8 compliance:  https://travis-ci.org/github/ioos/compliance-checker/jobs/672560182

It allows for failure, but to my knowledge it's running across the entire code base, so we'd have to do some real refactoring of the code base if we didn't want that test to fail and to provide valuable input.

Thus, until we push our code through something like `black` and refactor, I'd prefer to handle this on the coding side.

Reply:
Not sure what you had in mind for refactoring but you can get the code base a pretty good ways to pep8 compliant with autopep8. I am definitely not an autopep8 expert but the couple times I've used it I was happy with the results. 

```
find ./compliance_checker -name \*.py -exec autopep8 --recursive --aggressive --in-place {} +
```

Reply:
@ocefpaf might know more about some of the tooling here.  Perhaps we should open a new issue to address coding standards/PEP8, etc.

Reply:
Do you want to use `pre-commit` hooks do to this? I can setup GH actions as a CI and those who have the pre-commit locally will get the code adjustments "for free." I'll open a PR to demonstrate that.